libminifi/include/core/Processor.h

```diff
@@ -239,12 +242,21 @@ class Processor : public Connectable, public ConfigurableComponent,
   // Trigger the Processor even if the incoming connection is empty
   std::atomic<bool> _triggerWhenEmpty;
 
- private:
+  //! obtainSite2SiteProtocol for use
+  std::shared_ptr<Site2SiteClientProtocol> obtainSite2SiteProtocol(std::string host, uint16_t sport, uuid_t portId);
```

Comment:
@benqiu2016 Can you make a ticket to move this out of Processor? It should be a separate component. If you make the ticket I'll make this particular change.